src/components/learner-credit-management/data/hooks/useBudgetContentAssignments.js

```diff
@@ -76,15 +81,45 @@ const useBudgetContentAssignments = ({
       };
       applyFiltersToOptions(args.filters, options);
       applySortByToOptions(args.sortBy, options);
+
+      const argsCopy = {
+        pageIndex: args?.pageIndex,
+        pageSize: args?.pageSize,
+        filters: args?.filters,
+        sortBy: args?.sortBy,
+      };
+      /* This logic in conjunction with useRef is being used to prevent track events
+      from being called when the page re-renders without the specifically selected
+      arguments (argCopy) being changed */
```

Comment:
nit: might recommend putting the comment above `argsCopy` as I personally would first have a question about why `argsCopy` is created before seeing how its used below.

Reply:
Done 👍🏽